src/Microsoft.ML.Transforms/NAIndicatorTransform.cs

```diff
@@ -385,5 +475,272 @@ private void FillValues(int srcLength, ref VBuffer<bool> dst, List<int> indices,
+                return result;
+            }
+
+            protected override Delegate MakeGetter(IRow input, int iinfo, out Action disposer)
+            {
+                Host.AssertValue(input);
+                Host.Assert(0 <= iinfo && iinfo < _infos.Length);
+                disposer = null;
+
+                if (!_infos[iinfo].TypeSrc.IsVector)
+                    return ComposeGetterOne(input, iinfo);
+                return ComposeGetterVec(input, iinfo);
+            }
+
+            /// <summary>
+            /// Getter generator for single valued inputs.
+            /// </summary>
+            private ValueGetter<bool> ComposeGetterOne(IRow input, int iinfo)
+                => Utils.MarshalInvoke(ComposeGetterOne<int>, _infos[iinfo].TypeSrc.RawType, input, iinfo);
+
+            private ValueGetter<bool> ComposeGetterOne<T>(IRow input, int iinfo)
+            {
+                var getSrc = input.GetGetter<T>(ColMapNewToOld[iinfo]);
+                var src = default(T);
+                var isNA = (RefPredicate<T>)_isNAs[iinfo];
+
+                ValueGetter<bool> getter;
+
+                return getter =
+                    (ref bool dst) =>
+                    {
+                        getSrc(ref src);
+                        dst = isNA(ref src);
+                    };
+            }
+
+            /// <summary>
+            /// Getter generator for vector valued inputs.
+            /// </summary>
+            private ValueGetter<VBuffer<bool>> ComposeGetterVec(IRow input, int iinfo)
+                => Utils.MarshalInvoke(ComposeGetterVec<int>, _infos[iinfo].TypeSrc.ItemType.RawType, input, iinfo);
+
+            private ValueGetter<VBuffer<bool>> ComposeGetterVec<T>(IRow input, int iinfo)
+            {
+                var getSrc = input.GetGetter<VBuffer<T>>(ColMapNewToOld[iinfo]);
+                var isNA = (RefPredicate<T>)_isNAs[iinfo];
+                var val = default(T);
+                var defaultIsNA = isNA(ref val);
+                var src = default(VBuffer<T>);
+                var indices = new List<int>();
+
+                ValueGetter<VBuffer<bool>> getter;
+
+                return getter =
+                    (ref VBuffer<bool> dst) =>
+                    {
+                        // Sense indicates if the values added to the indices list represent NAs or non-NAs.
+                        bool sense;
+                        getSrc(ref src);
+                        _parent.FindNAs(ref src, isNA, defaultIsNA, indices, out sense);
+                        _parent.FillValues(src.Length, ref dst, indices, sense);
+                    };
+            }
+        }
+    }
+
+    public sealed class NAIndicatorEstimator : IEstimator<NAIndicatorTransform>
+    {
+        private readonly IHost _host;
+        private readonly NAIndicatorTransform.ColumnInfo[] _columns;
+
+        public NAIndicatorEstimator(IHostEnvironment env, string name, string source = null)
```

Comment:
>source [](start = 78, length = 6)

don't use `name` and `source`, use `input` and `output`. See the other similar estimators/transformers